### ISSUE TITLE
Unity+Vuforia setup needs extra steps

### ISSUE THREAD

**Issue by MRzr** [jiulongw/swift-unity#48]
Hi,

thanks for your effort to make swift with unity easier to make it work.

When using Vuforia, there is some extra work that needs to be done in order to work as expected. In my case ImageTarget was not working. Xcode error was **Dataset 'XYZ' could not be loaded and cannot be activated.**

Unity creates folder Data which is not copied by run scripts. I solved a problem by copying Data folder to project Unity folder and adding folder reference to Vuforia (Data/Raw/Vuforia) to my project.

I don't know if this is too specific problem to be solved by your script. If it is this issue can at least help someone who will needs to work with Vuforia.

**Comment by jiulongw** [comment]
I don't know what is special in Vuforia that Data folder has to be added to project.  The script does copy Data folder, but directly to the app folder because by default the Data folder are only used in run time.

> echo "Syncing data from $UNITY_IOS_EXPORT_PATH..."
> rm -rf "$TARGET_BUILD_DIR/$PRODUCT_NAME.app/Data"
> cp -Rf "$UNITY_IOS_EXPORT_PATH/Data" "$TARGET_BUILD_DIR/$PRODUCT_NAME.app/Data"

Do you mean adding Data folder into project solves the "Dataset 'XYZ' could not be loaded and cannot be activated." problem?  Would you mind compare the build output to see what's different so we can adjust the script to handle Vuforia?

**Comment by MRzr** [comment]
I guess it's because Vuforia searches for resources contained in that folder and try to load it from main bundle. Is cp command is works exactly as what Xcode does in Copy Bundle Resource step? 

[Build accolade_2018-02-27T13-35-20 : With Vuforia.txt](https://github.com/jiulongw/swift-unity/files/1762663/Build.accolade_2018-02-27T13-35-20.With.Vuforia.txt)
[Build accolade_2018-02-27T13-41-28.txt](https://github.com/jiulongw/swift-unity/files/1762664/Build.accolade_2018-02-27T13-41-28.txt)

**Comment by xh4010** [comment]
Hi
I also encountered a similar problem, Vuforia ARCamera can not work, the problem is repeated:
unity2017.3.1f1 + xcode9.2
1. Open jiulongw's demo (./unity), demo scene Add ARCamera (GameObject> Vuforia> ARCamera), delete MainCamera.
2. Build setting:
set camera usage description to demoApp
set target minimum ios version to 9.0
3. Build to tmp folder
4. open xcode project (./xcode/DelayApp.xcodeproj), set Enable Bitcode for build setting to NO.
5. Select Device (Need a real machine), command + B, Build succeeded.
6. command + R, run for real machine, app crash.
![image](http://p35p8oneg.bkt.clouddn.com/20180307152041188145163.png)

**Comment by xh4010** [comment]
vuforia ARcamera not work bug is solved,  refer [issues #34](https://github.com/jiulongw/swift-unity/issues/34)

> Try going into your Player settings -> Other Settings and
>  · turn off Auto Graphics API,
>  · delete 'Metal' from the Graphics API list.

**Comment by mayntos** [comment]
@MRzr what were the steps you performed to add the Vuforia folder reference to your project? We are encountering a similar issue where ImageTarget is not working.

**Comment by jiulongw** [comment]
@mayntos try to add Data folder into Xcode project.  Xcode might be doing extra steps other than simply copy content to app output.

**Comment by piotrdebosz** [comment]
@mayntos I've tested this multiple times. The only thing other than following steps in instruction than you need to do is to import Data folder into project. Steps:

1. Open Unity folder (one you've already copied into your Xcode project) in Finder
2. Copy Data folder from folder generated by Unity.
3. In your Xcode project select Unity folder, press right click and choose "Add files to PROJECT_NAME"
4. Choose Data/Raw/Vuforia folder. Press Options button and select "Create folder references" in "Added folders" option.
5. Make sure that Vuforia folder is in Target Settings -> Build Phases -> Copy Bundle Resources.

You don't need to do it every time you generate project in Unity. Do it only when you change targets in Unity. 

Let me know if you have any problems.

**Comment by metamorphious** [comment]
I'm trying to build the demo with Vuforia but keep getting this error:

```
cameraDeviceStartCamera
ERROR/AR(545) 2018-04-31 22:41:15: VideoBackgroundConfig with screen size of zero received, skipping config step
```

Did all of the things mentioned above (turn off bitcode, turn off metal, copy and reference Data folder).
Any ideas?

**Comment by MRzr** [comment]
@metamorphious do you encounter black screen?  its iOS 11.3 & vuforia issue. it is sloved in vuforia version 7.1.34 you can update from unity 2018.1.

**Comment by metamorphious** [comment]
yeah had black screens. Tried with Unity 2018.1 beta but now the unity storyboard isnt found anymore:
`'Storyboard (<UIStoryboard: 0x1d066ce00>) doesn't contain a view controller with identifier 'unitySplashStoryboard''`

**Comment by MRzr** [comment]
@metamorphious i think it is because XcodePostBuild script is not yet compatible with 2018.1
try to open new issue becuase your problem is not relevant with with this issue
